### DISCUSSION TITLE
skywalking java agent takes too many memory

### DISCUSSION THREAD

**Discussion by SimonLiuhd** [apache/skywalking#7955] [Q&A]
Hi all,

I have a tomcat 8 application, when configured the SkyWalking java agent in the catalina.sh, the JVM will take 16G memory； while removed the agent related configs，the memory downed to about 3G.
And I the another application which have the same SkyWalking java agent configurations, but does not take much memory.
I cannot figure out the cause.

Any thoughts?

**Comment by wu-sheng** [comment]
By doing a memory analysis, you should take the memory dump and check what is the leak(or whether there is a leak) first.
